src/pyodide/internal/workers.py

```diff
@@ -96,6 +95,10 @@ def replace_body(self, body: Body) -> "FetchResponse":
         js_resp = js.Response.new(b, self.js_response)
         return FetchResponse(js_resp.url, js_resp)
 
+    async def blob(self) -> "Blob":
+        self._raise_if_failed()
+        return _js_value_to_py(await self.js_object.blob())
```

Comment:
Unless it's possible for `blob()` to return objects other than `Blob`s, I'd think we should do:
```suggestion
        return Blob(await self.js_object.blob())
```